File: FRun_demo/FRun/BaseViewController.m
//
//  BaseViewController.m
//  FRun
//
//  Created by noise on 2016/11/7.
//  Copyright © 2016年 noisecoder. All rights reserved.
//

#import "BaseViewController.h"

@interface BaseViewController ()

@end

@implementation BaseViewController

-(void)viewWillAppear:(BOOL)animated{
    [super viewWillAppear:animated];
    self.navigationItem.hidesBackButton = YES;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    // Do any additional setup after loading the view.
}

//布局左边 设置返回图片
-(void)createBarLeftWithImage:(NSString *)imageName{
    
    if (imageName==nil) {
        return;
    }
    
    if ([imageName length]==0) {
        return;
    }
    
    UIButton *btnb = [UIButton buttonWithType : UIButtonTypeCustom];
    
    [btnb setImage:[UIImage imageNamed:imageName] forState:UIControlStateNormal];
    [btnb sizeToFit];
    
    // btnb.showsTouchWhenHighlighted=YES;
    [btnb addTarget:self action:@selector(showLeft:) forControlEvents:UIControlEventTouchUpInside];
    UIBarButtonItem *ubar=[[UIBarButtonItem alloc] initWithCustomView :btnb];
    self.navigationItem.leftBarButtonItem = ubar;
    
}

-(void)showLeft:(UIButton *)sender{
    [self.navigationController popViewControllerAnimated:YES];
}


//设置右边导航按钮字体和颜色
-(void)createRightTitle:(NSString *)title  titleColor:(UIColor*)titleColor{
    
    UIButton *btnb = [UIButton buttonWithType : UIButtonTypeCustom];
    if (title.length>3) {
        btnb.frame = CGRectMake (0, 0, 80, 44);
    }
    else
    {
        btnb.frame = CGRectMake (0, 0, 44, 44);
        
    }
    btnb.titleLabel.font = [UIFont boldSystemFontOfSize:16.0];
    
    [btnb setTitle:title forState:UIControlStateNormal];
    [btnb setTitleColor:titleColor forState:UIControlStateNormal];
    [btnb setTitleColor:[UIColor grayColor] forState:UIControlStateHighlighted];
    btnb.titleLabel.textAlignment = NSTextAlignmentRight;
    [btnb addTarget:self action:@selector(showRight:) forControlEvents:UIControlEventTouchUpInside];
    UIBarButtonItem *ubar=[[UIBarButtonItem alloc] initWithCustomView :btnb];
    self.navigationItem.rightBarButtonItem = ubar;
    
}
//设置右边导航按钮图片
-(void)createBarRightWithImage:(NSString *)imageName{
    if (imageName==nil) {
        
        return;
    }
    
    if ([imageName length]==0) {
        return;
    }
    
    UIButton *remindBtn = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 30, 24)];
    //    [_remindBtn setBackgroundImage :[UIImage imageNamed:imageName] forState:UIControlStateNormal];
    [remindBtn setImage :[UIImage imageNamed:imageName] forState:UIControlStateNormal];
    
    [remindBtn addTarget:self action:@selector(showRight:) forControlEvents:UIControlEventTouchUpInside];
    UIBarButtonItem *ubar=[[UIBarButtonItem alloc] initWithCustomView :remindBtn];
    self.navigationItem.rightBarButtonItem = ubar;
    
}


-(void)showRight:(UIButton *)sender{
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: project/zuler_ios/BaseViewController.swift
//
//  BaseViewController.swift
//  zuler_ios
//
//  Created by Admin on 2021/8/16.
//

import UIKit
import RxSwift
import RxCocoa

class BaseViewController: QMUICommonViewController {
    
    open lazy var bag = DisposeBag()
    
    open lazy var leftBtn : UIButton = {
        let btn = UIButton.init(type: .custom)
        btn.titleLabel?.font = UIFont.systemFont(ofSize: 14)
        btn.setTitleColor(UIColor.white, for: .normal)
        btn.contentHorizontalAlignment = .left
        btn.frame = CGRect.init(x: 0, y: 0, width: 25, height: 25)
        btn.addTarget(self, action: #selector(leftButtonAction(_:)), for: .touchUpInside)
        self.navigationItem.leftBarButtonItem = UIBarButtonItem.init(customView: btn)
        
        return btn
    }()

    open lazy var rightBtn:UIButton = {
        let btn = UIButton.init(type: .custom)
        btn.titleLabel?.font = UIFont.systemFont(ofSize: 14)
        btn.setTitleColor(UIColor.hex_181818, for: .normal)
        btn.setTitleColor(UIColor.hex_999999, for: .disabled)
        btn.contentHorizontalAlignment = .right
        btn.frame = CGRect.init(x: 0, y: 2.5, width: 60, height: 25)
        btn.addTarget(self, action: #selector(rightButtonAction(_:)), for: .touchUpInside)
        let v = UIView(frame: CGRect(x: 0, y: 0, width: 60, height: 30))
        v.addSubview(btn)
        self.navigationItem.rightBarButtonItem = UIBarButtonItem.init(customView: v)
        
        return btn
    }()
    
    override var preferredStatusBarStyle: UIStatusBarStyle{
        return .darkContent
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.view.backgroundColor = .white
        
        setNavBarBackgroundColor(.white)
        
        self.navigationController?.navigationBar.isTranslucent = true
        
        setDefaultBackItem()
        
        // Do any additional setup after loading the view.
    }
    
    func setNavBarBackgroundColor(_ bgColor : UIColor) {
        let barImg = UIImage.imageWithColor(bgColor)
        self.navigationController?.navigationBar.setBackgroundImage(barImg, for: .any, barMetrics: .default)

    }

    override func shouldCustomizeNavigationBarTransitionIfHideable() -> Bool {
        return true
    }
    
    override func forceEnableInteractivePopGestureRecognizer() -> Bool {
        return true
    }
    
    
    public func setDefaultBackItem() -> Void {
        if self.navigationController?.viewControllers.count ?? 1 > 1 {
            self.setNavLeftItemWithTitleAndImg("", withImgName: "back")
        }
        self.setupNavigationItems()
    }

    //  MARK: - 设置左边按钮的标题，图片
    /// 设置左边按钮的标题，图片
    func setNavLeftItemWithTitleAndImg(_ title:String,withImgName name:String) -> Void {

        self.leftBtn.setTitle(title, for: .normal)
        self.leftBtn.setImage(UIImage(named: name), for: .normal)
    }
    
    
    @objc func leftButtonAction(_ btn:UIButton) -> Void {
        
        self.navigationController?.popViewController(animated: true)
    }
    
    // MARK: - 右边按钮事件
    /// 右边按钮事件
    @objc func rightButtonAction(_ btn:UIButton) -> Void {
        
    }
    
    
    // 隐藏导航栏
    //    override func preferredNavigationBarHidden() -> Bool {
    //        return true
    //    }
    //
    //    override func navigationBarBarTintColor() -> UIColor? {
    //        return UIColor.hex_FFFFFF
    //    }
    //
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}
